### PR TITLE
Cockpit APIs & UI (stub)

### PR DESCRIPTION
**Pull request by andre8244**
- UI
- APIs to connect/disconnect to/from a NS8 cluster
- APIs to manage migration of applications (scripts to complete account provider migration are not implemented yet)

### CODE REVIEW

File: api/migration/update [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        /sbin/e-smith/signal-event -j nethserver-ns8-migration-migrate-app mattermost
+
+        if [ $? -gt 0 ]; then
+            set_migration_status $app_id "migrating"
+            error "EventFailed" "See /var/log/messages"
+        fi
+
+        set_migration_status $app_id "migrated"
+    fi
+fi
```
[the remark this answers is not in the document]
Comment (andre8244) [reply api/migration/update:133]: Yes, I think we can. We can avoid the repeated IFs but we'll need to add an additional IF to manage Nextcloud virtual host

File: api/migration/update [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        if [ $? -gt 0 ]; then
+            set_migration_status $app_id "migrating"
+            error "EventFailed" "See /var/log/messages"
+        fi
+
+        set_migration_status $app_id "migrating"
+    elif [[ "$action" == "finish" ]]; then
+        set_migration_status $app_id "syncing"
+        /sbin/e-smith/signal-event -j nethserver-ns8-migration-migrate-app mattermost
```
[the remark this answers is not in the document]
Comment (andre8244) [reply api/migration/update:124]: The only (but important) purpose of the event is to darken Cockpit frame and show a notification while the action is running. We need this to give feedback to the user and inform it that something is happening under the hood

File: api/migration/update [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# along with NethServer.  If not, see COPYING.
+#
+
+. /usr/libexec/nethserver/api/lib/helper_functions
+
+set_migration_status() {
+    app_id=$1
+    status=$2
+    ns8_leader=$(/sbin/e-smith/config getprop ns8 Host)
+    /usr/bin/mkdir -p /var/lib/nethserver/agent/${ns8_leader}
```
Comment (DavidePrincipi) [inline api/migration/update:29]: Is it necessary to create a state directory for each cluster?
I'd prefer to create/reset always the same path in the login/logut script. What do you think?

Comment (DavidePrincipi) [reply api/migration/update:29]: I'm afraid multiple and partial migrations are hard to manage, also for the sysadmin. 

Furthermore 
- a module can still be in ns7 or migrated elsewhere. Once migrated we must remember its state and switching the state dir implies forget its state.
- ns7 cannot be disconnected from ns8 after account provider has been migrated.

**Review by DavidePrincipi — APPROVED** [review]

**Review by edospadoni — APPROVED** [review]

**Pull request merged**